### PR TITLE
Separate ARTProtocolMessage private parts.

### PR DESCRIPTION
**Pull request by tcard**
To adjust the public API to the spec.

### CODE REVIEW

**Comment by ricardopereira** [comment]
LGTM

File: ably-ios/ARTProtocolMessage+Private.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by Toni Cárdenas on 26/01/2016.
+//  Copyright (c) 2014 Ably. All rights reserved.
+//
+
+ART_ASSUME_NONNULL_BEGIN
+
+@interface ARTProtocolMessage ()
+
+@property (art_nullable, readwrite, strong, nonatomic) NSString *clientId;
```
Comment (mattheworiordan) [inline ably-ios/ARTProtocolMessage+Private.h]: Don't think a ProtocolMessage needs to have a clientId, http://docs.ably.io/client-lib-development-guide/features/#types

**Comment by mattheworiordan** [comment]
One comment, but otherwise good

File: ably-ios/ARTProtocolMessage+Private.h (hunk 1 above)
Comment (tcard) [inline ably-ios/ARTProtocolMessage+Private.h]: It is implemented kind of backwards. I can see the point, which is to store the ConnectionDetails in the ProtocolMessage itself, and just have the former point to the latter, but I think the other way around (ie. having a ProtocolMessage.connectionKey getter that points to ConnectionDetails.connectionKey if exists) . The API isn't affected, though (this is a private header).

Done at 7bef28a81c3be38f4c3f7dfd84d49e57cf88eac8.

**Comment by tcard** [comment]
PTAL

**Comment by mattheworiordan** [comment]
:+1:

**Comment by ricardopereira** [comment]
LGTM and tests are passing :+1:

**Pull request merged**